beancount/query/query_compile.py

```diff
@@ -204,23 +193,21 @@ class EvalMul(EvalBinaryOp):
 
     def __init__(self, left, right):
         f = lambda x, y: (x * y)
+        resultType = right.dtype
         if left.dtype != right.dtype:
             if left.dtype == amount.Amount:
-               dtype = left.dtype
-            else:
-               dtype = right.dtype
-        super().__init__(f, left, right, dtype)
+               resultType = left.dtype
```

Comment:
Multiplying a `Decimal` by an `Amount` is permitted, but the computation of the return type here would get it wrong. Also, we use snake case and not camel case for variables.

Reply:
I'm not seeing the issue. If Left=Decimal, and Right=Amount, then the default right.dtype (Amount) will be amount and this will work. If Left=Amount and Right=Decimal, then then left.dtype == amount.Amount will be true and resultType will be left.dtype which is Amount. Am I missing something.

I'll switch to snake_case on the next commit.

Reply:
In this context `Decimal` and `Amount` are not the only two possible dtypes.

Reply:
I think this is addressed in my latest version.